x-pack/plugins/fleet/server/services/epm/archive/extract.ts

```diff
@@ -19,14 +21,22 @@ export async function untarBuffer(
 ): Promise<unknown> {
   const deflatedStream = bufferToStream(buffer);
   // use tar.list vs .extract to avoid writing to disk
-  const inflateStream = tar.list().on('entry', (entry: tar.FileStat) => {
-    const path = entry.header.path || '';
+  const inflateStream = tar.list().on('entry', (entry) => {
+    const path = entry.path || '';
     if (!filter({ path })) return;
-    streamToBuffer(entry).then((entryBuffer) => onEntry({ buffer: entryBuffer, path }));
+    streamToBuffer(entry as unknown as NodeJS.ReadableStream).then((entryBuffer) =>
+      onEntry({ buffer: entryBuffer, path })
+    );
   });
 
-  return new Promise((resolve, reject) => {
-    inflateStream.on('end', resolve).on('error', reject);
+  return new Promise<void>((resolve, reject) => {
+    finished(inflateStream, (err) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve();
+      }
+    });
```

Comment:
According to the types `inflateStream` will never emit an `error` event. I'm not sure if we can 100% trust this, but using `finished` instead of listening for `end` is normally better practice anyway, and by doing so we actually also get any `error` events that might occur (without having to work around TS telling us it's not happening).